aws/convert_types.go

```diff
@@ -311,6 +311,15 @@ func TimeValue(v *time.Time) time.Time {
 	return time.Time{}
 }
 
+// Int64TimeValue converts an int64 pointer to a time.Time value or
+// time.Time{} if the pointer is nil.
+func Int64TimeValue(v *int64) time.Time {
```

Comment:
Since i don't think there is any requirement required for services to return a specific resolution of time when used in number format I think having a set expectation would be the best approach.

Milliseconds would provide the most general usage. The downside to this would be that i think it would defeat the purpose of this utility for second based int64 fields as the user would need to dereference the `*int64` value shift for milliseconds, and grab the pointer again before using the utility. This is where the idea of resolution specific methods came from.